classes/db/DAO.inc.php

```diff
@@ -583,4 +590,75 @@ function formatDateToDB($date, $defaultNumWeeks = null, $acceptPastDate = true)
 			return null;
 		}
 	}
+
+	/**
+	 * Retrieves a SELECT statement without the SELECT and ORDER BY clauses for optimization purposes
+	 * @return array The SQL query at the index 0 and the updated parameters at the index 1
+	 */
+	private static function _optimizeCountQuery(string $sql, array $params): array {
+		$findTopLevelExpression = static function (string $s, string $expression, int $index, ?int &$foundParams = null): int {
```

Comment:
If you think it's ok to keep this code, I can leave some comments to make it more reader friendly. Otherwise, I'll just save somewhere in my gists 😅